server/src/main/java/org/opensearch/action/get/GetRequest.java

```diff
@@ -163,6 +164,12 @@ public GetRequest preference(String preference) {
         return this;
     }
 
+    @Override
+    public String index() {
+        return super.index();
```

Comment:
This can return null. Is it okay for implements of DocRequest to return null here?

Reply:
In the case of GetRequest its guaranteed to be non-null because of the validate() method. I would imagine that this new category of request requires this method to return non-null.

Take for example an [IndexMonitorRequest](https://github.com/opensearch-project/common-utils/blob/main/src/main/kotlin/org/opensearch/commons/alerting/action/IndexMonitorRequest.kt#L12) (currently a subclass of ActionRequest), I would like to categorize this under DocRequest instead as it results in [index a doc](https://github.com/opensearch-project/alerting/blob/main/alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt#L499-L514) into a system index for the alerting plugin.

Reply:
I applied this change, but I think its redundant since SingleShardRequest has a `validateNonNullIndex` method...I'm confused why index is even marked `@Nullable` in that class.

For implementers of DocRequest I don't think there's a way to enforce that they return a non-null value though they should do so conventionally.

Reply:
It may be redundant at runtime, but the check will make sure the signatures align and make static analysis tools happy and make it clear to any future refactorings that this must return non-null unlike the parent method.